### PR TITLE
Rib 70 implement manage users of interest page

### PR DESCRIPTION
**Pull request by dondowoldo**
### Invitations / Applications page
- Invites directly assign a role to the person that is being invited. Maybe in the future we would want to have some notification to the user that he has been invited and he gets to decide if he wishes to join or not.

- New methods that replaced old ones have tests except for DTO fetching methods in InterestService, where tests will be added after merging RIB-37

### CODE REVIEW

File: src/main/java/it/rate/webapp/controllers/InterestAdminController.java [hunk 1]
```diff
@@ -58,15 +58,20 @@ public String editInterest(
...
     Optional<Interest> optInterest = interestService.findInterestById(interestId);
     if (optInterest.isEmpty()) {
       throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Interest not found");
     }
-    model.addAttribute("usersByRoles", manageInterestService.getUsersByRole(interestId));
+    if (principal != null) {
+      model.addAttribute(
+          "loggedUser",
+          userService
+              .findByEmail(principal.getName())
```
Comment (vdrfg) [inline src/main/java/it/rate/webapp/controllers/InterestAdminController.java:71]: Based on all Matej's comments, I guess this should be getByEmail. :P

File: src/main/java/it/rate/webapp/controllers/InterestAdminController.java [hunk 2]
```diff
@@ -79,8 +84,45 @@ public String removeUser(@PathVariable Long interestId, @PathVariable Long userI
...
+          throws BadRequestException {
+    Optional<Interest> optInterest = interestService.findInterestById(interestId);
+    if (optInterest.isEmpty()) {
+      throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Interest not found");
+    }
+    if (principal != null) {
+      model.addAttribute(
+              "loggedUser",
+              userService
+                      .findByEmail(principal.getName())
```
Comment (vdrfg) [inline src/main/java/it/rate/webapp/controllers/InterestAdminController.java:105]: And this one too?

File: src/main/java/it/rate/webapp/controllers/InterestAdminController.java (hunk 1 above)
Comment (dondowoldo) [reply src/main/java/it/rate/webapp/controllers/InterestAdminController.java]: Done. Also same done for retrieving interest, where failed retrieve scenario is already covered in Preauthorize method

**Review by vdrfg — APPROVED** [review]

**Review by BenediktVitek — APPROVED** [review]

File: src/main/java/it/rate/webapp/controllers/InterestAdminController.java [hunk 3]
```diff
@@ -30,11 +28,7 @@ public class InterestAdminController {
   @GetMapping("/edit")
   @PreAuthorize("hasAnyAuthority(@permissionService.manageCommunity(#interestId))")
   public String editInterestPage(@PathVariable Long interestId, Model model, Principal principal) {
-    Optional<Interest> interest = interestService.findInterestById(interestId);
-    if (interest.isEmpty()) {
-      throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Interest not found");
-    }
-    model.addAttribute("interest", interest.get());
+    model.addAttribute("interest", interestService.getById(interestId));
```
Comment (MatejFrnka) [inline src/main/java/it/rate/webapp/controllers/InterestAdminController.java:31]: is it 100% certain that the interest will exist? If not, the previous version with optional was better

File: src/main/java/it/rate/webapp/services/InterestService.java [hunk 4]
```diff
@@ -27,12 +30,15 @@ public Optional<Interest> findInterestById(Long id) {
     return interestRepository.findById(id);
   }
 
+  public Interest getById(Long id) {
```
Comment (MatejFrnka) [inline src/main/java/it/rate/webapp/services/InterestService.java:33]: this is related to my comment above, if you end up changing it back, remove this as well.

File: src/test/java/it/rate/webapp/controllers/InterestAdminControllerIntegrationTest.java [hunk 5]
```diff
@@ -128,8 +128,6 @@ void editUsersReturnsForbiddenForVoter() throws Exception {
         .perform(get("/interests/" + interestId + "/admin/users"))
         .andExpect(status().isForbidden())
         .andReturn();
-
-    verify(manageInterestService, times(0)).getUsersByRole(interestId);
```
Comment (MatejFrnka) [inline src/test/java/it/rate/webapp/controllers/InterestAdminControllerIntegrationTest.java:132]: why was this removed, it is important part of the test that verifies that forbidden users really didn't get access?

File: src/main/java/it/rate/webapp/controllers/InterestAdminController.java (hunk 3 above)
Comment (dondowoldo) [reply src/main/java/it/rate/webapp/controllers/InterestAdminController.java:31]: It should be 100%, as the manageCommunity method inside Preauthorize annotation gets executed first, and checks if Interest with given id exists and if not, throws an exception.

File: src/test/java/it/rate/webapp/controllers/InterestAdminControllerIntegrationTest.java (hunk 5 above)
Comment (dondowoldo) [reply src/test/java/it/rate/webapp/controllers/InterestAdminControllerIntegrationTest.java:132]: getUsersByRole method no longer exist, but indeed a verification is missing here for the new method. I will update this.

**Review by MatejFrnka — APPROVED** [review]
